src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    public static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int SCALE_FACTOR = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int bucketNumber = getBucketNumber(key);
+        if (table[bucketNumber] == null) {
+            table[bucketNumber] = new Node<>(key, value, null);
+        } else {
+            Node<K, V> bucketNode = table[bucketNumber];
+            do {
+                if (Objects.equals(bucketNode.key, key)) {
+                    bucketNode.value = value;
+                    return;
+                } else if (bucketNode.next == null) {
+                    bucketNode.next = new Node<>(key, value, null);
+                    break;
+                }
+                bucketNode = bucketNode.next;
+            } while (bucketNode != null);
+        }
+        size++;
+        if (table.length * LOAD_FACTOR < size) {
```

Comment:
I'm not sure if I understand correctly this line, but shouldn't we do resizing if the size is bigger than the threshold, not less?

Reply:
agree